### PR TITLE
update Java/11 to 11.0.16 and Java/17 to 17.0.4

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)
I didn't bother with Java 12-16, since they aren't recommended for use anyway. https://whichjdk.com/ suggests going to 17 next.

### CODE REVIEW

File: easybuild/easyconfigs/j/Java/Java-11.eb [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
+Java applications on desktops and servers."""
 
 toolchain = SYSTEM
 
-dependencies = [
-    ('Java', {'arch=x86_64': '%(version)s.0.2',
-              'arch=AArch64': '%(version)s.0.8-aarch64',
-              'arch=POWER': '%(version)s.0.6-ppc64le'}),
-]
+dependencies = [('Java', '%(version)s.0.4')]
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/j/Java/Java-11.eb:12]:
```suggestion
dependencies = [('Java', '%(version)s.0.16')]
```

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.5, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/f4cdefea4f79222f69451b1b6aa4065b for a full test report.

**Comment by Micket** [comment]
@Flamefire could you test on PPC?

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
bdw-opensuse-154 - Linux openSUSE Leap 15.4, x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 510.85.02, Python 3.6.15
See https://gist.github.com/a76706e1bb78f37bb9c35322f100b9e0 for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
ugh, line length, this will get ugly

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.5, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/b4dfa58236a2fc0d5e99d9a1a15529de for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
easybuild2.novalocal - Linux CentOS Stream 8, POWER, IBM pSeries (emulated by qemu) (power9le), Python 3.6.8
See https://gist.github.com/d64255c2963c944d2b5c78600d81bf5d for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
fair-mastodon-c6g-2xlarge-0001 - Linux Rocky Linux 8.5, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/f5eb717380582267cc743324180f050e for a full test report.

**Review by boegel — APPROVED** [review]
lgtm, tested on `x86_64`, `aarch64`, `ppc64le`

**Comment by boegel** [comment]
Going in, thanks @Micket!

**Pull request merged**